rust/src/main.rs

```diff
@@ -205,27 +208,60 @@ fn read_block_range(pkg: &Package, module_name: &str) -> Result<(i64, u64), anyh
     return Ok((start, stop));
 }
 
-async fn read_package(input: &str) -> Result<Package, anyhow::Error> {
+async fn read_package(
+    input: &str, 
+    module_name: &str, 
+    filter_query: Option<String>,
+) -> Result<Package, Error> {
     let mut mutable_input = input.to_string();
 
     let val = parse_standard_package_and_version(input);
     if val.is_ok() {
-        let package_and_version = val.unwrap();
+        let package_and_version = val?;
         mutable_input = format!(
             "{}/v1/packages/{}/{}",
             REGISTRY_URL, package_and_version.0, package_and_version.1
         );
     }
 
-    if mutable_input.starts_with("http") {
-        return read_http_package(&mutable_input).await;
+    let mut package = if mutable_input.starts_with("http") {
+        read_http_package(&mutable_input).await
+    } else {
+        // Assume it's a local file
+        let content = std::fs::read(&mutable_input)
+            .context(format_err!("read package from file '{}'", mutable_input))?;
+        Package::decode(content.as_ref()).context("decode command")
+    }?;
+    
+    if let Some(filter_query) = filter_query {
+        println!("Adding block filter '{}'", filter_query);
+
+        // Find the module by name and apply the block filter
+        if let Some(modules) = &mut package.modules {
+            if let Some(module) = modules.modules.iter_mut().find(|m| m.name == module_name) {
+                module.inputs[0].input = Some(Input::Params(Params {
+                    value: filter_query.to_string(),
+                }));
+            }
+        }
+        Ok(package)
+    } else {
+        Ok(package)
     }
+}
 
-    // Assume it's a local file
-    let content = std::fs::read(&mutable_input)
-        .context(format_err!("read package from file '{}'", mutable_input))?;
-    Package::decode(content.as_ref()).context("decode command")
+fn read_module_and_filter(third_arg: String) -> (String, Option<String>) {
+    if third_arg.contains(":") {
+        // Example: filtered_events:"(type:transfer)"
+        (
+            third_arg.split_once(":").unwrap().0.to_string(),
+            Some(third_arg.split_once(":").unwrap().1.to_string()),
+        )
+    } else {
+        (third_arg, None)
+    }
 }
```

Comment:
```suggestion
fn read_module_and_filter(input: &str) -> (&str, Option<&str>) {
    match input.split_once(":") {
        Some((key, value)) => (key, Some(value)),
        None => (input, None),
    }
}
```

Adjust the tests and calling code.

Reply:
Updated code. Thanks for reviewing!